polyfills/Map/polyfill.js

```diff
@@ -0,0 +1,124 @@
+(function(global) {
+
+
+	// Deleted map items mess with iterator pointers, so rather than removing them mark them as deleted. Can't use undefined or null since those both valid keys so use a private symbol.
+	var undefMarker = {};
```

Comment:
I didn't use Symbol because private symbols are not really private, so a basic empty object seems better to me.

https://mail.mozilla.org/pipermail/es-discuss/2014-January/035604.html